Ficzyr/moderate photos

lasgra opened the pull request:
Cool little small tiny feature 
![image](https://github.com/user-attachments/assets/7f52e6c2-1f81-43d1-86b2-616b2753325d)

lasgra committed:
moderating-feature

lasgra commented on src/app/services/api/account/account-pictures.service.ts:
```diff
@@ -1,8 +1,8 @@
 import { Injectable } from '@angular/core';
```
to jest ten plik co zmieniłem importy

Majkoo commented on src/app/services/api/account/auth.service.ts:
```diff
@@ -68,6 +68,15 @@ export class AuthService {
     return this.getJwtData() != null;
   }
 
+  get isAdmin(): boolean {
```
name doesnt match behavior

Majkoo replied:
my preference is to have two methods: isAdmin, isModerator. Then in corresponding places its just if (isAdmin || isModerator) that is just readable

lasgra replied:
changed name to isModerator, it's well fitted because I don't need to make a ton of fixes, and technically admin is also a moderator

Majkoo replied:
no

Majkoo commented on src/app/shared/components/query-modal/viewPicture/view-picture-modal.component.html:
```diff
@@ -20,6 +20,7 @@
             <div class="flex gap-3 text-2xl icon-dark">
               <pp-like-btn [(pic)]="pic" />
               <span class="icon-share"></span>
+              <pp-mod-btn [pic]="pic"></pp-mod-btn>
```
name doesnt match purpose

lasgra replied:
it matches purpose very well

lasgra committed:
added settings functionality and other things

lasgra commented:
Turns out it wasn't little feature,

![image](https://github.com/user-attachments/assets/06a02d12-3d57-4b23-972c-b4699884d83e)
Added moderation buttons. One to delete image, and other to ban user, and automatically delete all of user's images

![image](https://github.com/user-attachments/assets/f506acb6-21f9-4275-a2a7-ffd6b2d45b88)
Added settings functionality, that uses localstorage to keep data, and settingsService to access them.

2 new buttons in settings, one to turn off or on moderation buttons, and one to show something special.

2 new components, settings-option for lightings speed adding new options to settings, and switch-btn which looks amazing,

and fixed few random things I found

Majkoo commented on src/app/services/api/picture/picture.service.ts:
```diff
@@ -35,6 +35,10 @@ export class PictureService {
   }
 
+  delete(id: string): Observable<String>{
+    return this.httpClient.delete<String>(`${environment.apiUrl}/picture/${id}`) 
```
uppercase String?

Majkoo commented on src/app/shared/components/setting-option/setting-option.component.ts:
```diff
@@ -0,0 +1,14 @@
+})
+export class SettingOptionComponent {
+  @Input({required: true}) text!: string;
+  @Input() changeFlipState: boolean = false;
```
value, switchValue, state,

Majkoo commented on src/app/shared/components/setting-option/setting-option.component.ts:
```diff
@@ -0,0 +1,14 @@
+export class SettingOptionComponent {
+  @Input({required: true}) text!: string;
+  @Input() changeFlipState: boolean = false;
+  @Output() switchFlipped: EventEmitter<boolean> = new EventEmitter<boolean>();
```
switched, valueChanged

Majkoo commented on src/app/shared/components/switch-btn/switch-btn.component.ts:
```diff
@@ -0,0 +1,16 @@
+export class SwitchBtnComponent {
+  @Input() flipped: boolean = false;
+  @Output() flip: EventEmitter<boolean> = new EventEmitter<boolean>();
+  ngOnInit(){
```
console log rm

Majkoo commented on src/app/shared/components/switch-btn/switch-btn.component.ts:
```diff
@@ -0,0 +1,16 @@
+  imports: [CommonModule]
+})
+export class SwitchBtnComponent {
+  @Input() flipped: boolean = false;
```
adjust naming

lasgra committed:
added shit michal wanted

Majkoo reviewed (changes requested):
splendid ✨
